### PR TITLE
[MD][Cred Mgr]PoC delete credentials

### PR DESCRIPTION
**Pull request by noCharger**
### Description

1. PoC delete credentials

Screen Shot 2022-07-11 at 12 59 36 PM

2. Resolve some UI feedbacks in #9 

### Issue
* Add Table route

### CODE REVIEW

File: src/plugins/credential_management/public/components/credential_table/credentials_table.tsx [hunk 1]
```diff
@@ -112,8 +114,46 @@ export const CredentialsTable = ({ canSave, history }: Props) => {
...
+
+    return (
+      <EuiButton color="danger" iconType="trash" onClick={onClickDelete}>
+        Delete {selectedCredentials.length} Credentials
+      </EuiButton>
+    );
+  };
+
+  const onClickDelete = () => {
+    Promise.resolve(deleteCredentials(savedObjects.client, selectedCredentials)).then(function () {
```
Comment (seraphjiang) [inline src/plugins/credential_management/public/components/credential_table/credentials_table.tsx:148]: TODO: add validation logic
- prevent user to delete credential has been used in any datasource.

**Review by seraphjiang — APPROVED** [review]

File: src/plugins/credential_management/public/components/credential_table/credentials_table.tsx (hunk 1 above)
Comment (noCharger) [reply src/plugins/credential_management/public/components/credential_table/credentials_table.tsx:148]: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1876

**Comment by kristenTian** [comment]
Can we follow the convention and add **title** filed to credential saved object mapping? 


This can enable some logic and code reusability.  Thanks.

**Comment by noCharger** [comment]
> Can we follow the convention and add **title** filed to credential saved object mapping?
> 
> This can enable some logic and code reusability. Thanks.

If I'm not mistaken, the 'title' field is not provided by the SavedObjectsType interface as metadata. Although the management field 'getTitle' is defined in the SavedObjectsTypeManagementDefinition interface, overriding it allowed us to get the specific attribute directly. In stead of using `getTitle`, we can use `getCredentialName` function provided here - https://github.com/elastic-analytics/OpenSearch-Dashboards/blob/2.x/src/plugins/credential_management/server/saved_objects/credential_saved_object_type.ts#L21

Willing to know which existing part you'd like to reuse and let's discuss on how to reuse that. Thanks

**Pull request merged**